fontc/src/lib.rs

```diff
@@ -3174,6 +3174,74 @@ mod tests {
         assert_eq!(rvrn_replacement.lookup_list_indices(), [0, 1]);
     }
 
+    #[test]
+    fn glyphs_feature_variations() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        let result = TestCompile::compile_source("glyphs3/LibreFranklin-bracketlayer.glyphs");
+        let gsub = result.font().gsub().unwrap();
+        let lookup_list = gsub.lookup_list().unwrap();
+        assert_eq!(lookup_list.lookup_count(), 1);
+        let feature_list = gsub.feature_list().unwrap();
+        // by default, glyphs puts feature variations in rlig
+        let rlig_idx = feature_list
+            .feature_records()
+            .iter()
+            .position(|rec| rec.feature_tag() == "rlig")
+            .unwrap();
+        let rlig = feature_list.feature_records()[rlig_idx];
+        // base rlig feature has no lookups
+        assert!(rlig
+            .feature(feature_list.offset_data())
+            .unwrap()
+            .lookup_list_indices()
+            .is_empty());
+
+        let featvars = gsub.feature_variations().unwrap().unwrap();
+        let var_rec = featvars.feature_variation_records()[0];
+        let feature_substitution = var_rec
+            .feature_table_substitution(featvars.offset_data())
+            .unwrap()
+            .unwrap();
+
+        let rlig_sub_record = feature_substitution.substitutions()[0];
+
+        assert_eq!(rlig_sub_record.feature_index() as usize, rlig_idx);
+        let rlig_replacement = rlig_sub_record
+            .alternate_feature(feature_substitution.offset_data())
+            .unwrap();
+
+        assert_eq!(rlig_replacement.lookup_list_indices(), [0]);
+    }
```

Comment:
It's quite hard to read these tests. There's so much unwrapping and record fetching it dominates the code. Idk if we need helper functions or what, but my intuition is it should be possible to make it a bit more readable.

Reply:
the structure of the featurevariations table is one of the most baroque I've seen... split this out into a helper, at least.